api/v2_vulnerabilities.go

```diff
@@ -0,0 +1,310 @@
+//
+// Author:: Salim Afiune Maya (<afiune@lacework.net>)
+// Copyright:: Copyright 2022, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package api
+
+import "time"
+
+// v2VulnerabilitiesService is a service that interacts with the Vulnerabilities
+// endpoints from the Lacework APIv2 Server
+type v2VulnerabilitiesService struct {
+	client     *Client
+	Hosts      *v2HostVulnerabilityService
+	Containers *v2ContainerVulnerabilityService
+}
+
+func NewV2VulnerabilitiesService(c *Client) *v2VulnerabilitiesService {
+	return &v2VulnerabilitiesService{c,
+		&v2HostVulnerabilityService{c},
+		&v2ContainerVulnerabilityService{c},
+	}
+}
+
+// v2ContainerVulnerabilityService is a service that interacts with the APIv2
+// vulnerabilities endpoints for containers
+type v2ContainerVulnerabilityService struct {
+	client *Client
+}
+
+// SearchLastWeek returns a list of VulnerabilityContainer from the last 7 days
+func (svc *v2ContainerVulnerabilityService) SearchLastWeek() (VulnerabilitiesContainersResponse, error) {
+	var (
+		now    = time.Now().UTC()
+		before = now.AddDate(0, 0, -7) // 7 days from ago
+	)
+
+	return svc.Search(SearchFilter{
+		TimeFilter: &TimeFilter{
+			StartTime: &before,
+			EndTime:   &now,
+		},
+	})
+}
+
+// Search returns a list of VulnerabilityContainer from the last 7 days
+func (svc *v2ContainerVulnerabilityService) Search(filters SearchFilter) (
+	response VulnerabilitiesContainersResponse, err error,
+) {
+	err = svc.client.RequestEncoderDecoder(
+		"POST", apiV2VulnerabilitiesContainersSearch,
+		filters, &response,
+	)
+	return
+}
+
+// SearchAllPages iterates over all pages and returns a list of VulnerabilityContainer
+func (svc *v2ContainerVulnerabilityService) SearchAllPages(filters SearchFilter) (
+	response VulnerabilitiesContainersResponse, err error,
+) {
+	response, err = svc.Search(filters)
+	if err != nil {
+		return
+	}
+
+	var (
+		all    = []VulnerabilityContainer{}
```

Comment:
nitpick: 
```suggestion
		all    = []VulnerabilityContainer
```

append returns a nil slice so it's fine to have this as a nil slice rather than an empty slice.
Though if nothing is appended we'll return nil rather than empty VulnerabilityContainer slice. Would we rather return a nil value or empty slice?
The go wiki says prefer nil -> https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices
But we don't follow this practice consistently throughout the go-sdk.

Reply:
I think that suggestion is wrong. 
![Screen Shot 2022-02-11 at 8 19 54 AM](https://user-images.githubusercontent.com/5712253/153628533-fc43e9b1-daea-40fa-9a41-3d61cc2915d6.png)

Maybe: 
```suggestion
		all    []VulnerabilityContainer
```

Reply:
edit, equals was left in the suggestion.

Reply:
I wrote some tests and I don't see any change between these approaches, still, I am making the change.